core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,179 @@
+package cats
+package data
+
+import scala.annotation.tailrec
+import scala.collection.immutable.VectorBuilder
+import cats.std.vector._
+
+/**
+ * A data type which represents a single element (head) and a Vector
+ * (tail).  This can be used to represent a Vector which is guaranteed
+ * to not be empty.
+ */
+final case class NonEmptyVector[A] private (vector: Vector[A]) {
+
+  def unwrap: Vector[A] = vector
+
+  def head: A = vector.head
+
+  def tail: Vector[A] = vector.tail
+
+  /**
+   * remove elements not matching the predicate
+   */
+  def filter(f: A => Boolean): Vector[A] = vector.filter(f)
+
+  /**
+   * Append another NonEmptyVector to this
+   */
+  def combine(other: NonEmptyVector[A]): NonEmptyVector[A] = NonEmptyVector(vector ++ other.vector)
+
+  /**
+   * find the first element matching the predicate, if one exists
+   */
+  def find(f: A => Boolean): Option[A] = vector.find(f)
+
+  /**
+   * Check whether at least one element satisfies the predicate.
+   */
+  def exists(f: A => Boolean): Boolean = vector.exists(f)
+
+  /**
+   * Check whether all elements satisfy the predicate.
+   */
+  def forall(f: A => Boolean): Boolean = vector.forall(f)
+
+  /**
+   * Left-associative fold using f.
+   */
+  def foldLeft[B](b: B)(f: (B, A) => B): B =
+    vector.foldLeft(b)(f)
+
+  /**
+   * Right-associative fold using f.
+   */
+  def foldRight[B](lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    Eval.defer(Foldable[Vector].foldRight(vector, lb)(f))
+
+  /**
+    * Applies f to all the elements
+    */
+  def map[B](f: A => B): NonEmptyVector[B] =
+    NonEmptyVector(vector.map(f))
+
+  /**
+    *  Applies f to all elements and combines the result
+    */
+  def flatMap[B](f: A => NonEmptyVector[B]): NonEmptyVector[B] =
+    NonEmptyVector(vector.flatMap(a => f(a).vector))
+
```

Comment:
`reduce` and several other methods are available via the `Reducible` instance, but I agree that it would probably be nice to have some of these helpers on the class itself.